src/test/java/io/confluent/kafkarest/integration/JsonProducerTest.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package io.confluent.kafkarest.integration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.confluent.kafkarest.Versions;
+import io.confluent.kafkarest.entities.JsonProduceRecord;
+import io.confluent.kafkarest.entities.JsonTopicProduceRecord;
+import io.confluent.kafkarest.entities.PartitionOffset;
+import kafka.serializer.Decoder;
+import org.apache.kafka.common.errors.SerializationException;
+import org.junit.Before;
+import org.junit.Test;
+import scala.collection.JavaConversions;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+public class JsonProducerTest extends AbstractProducerTest {
+
+  private String topicName = "topic1";
+  private KafkaJsonDecoder decoder = new KafkaJsonDecoder();
+
+  public class KafkaJsonDecoder implements Decoder<Object> {
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    public Object fromBytes(byte[] bytes) {
+      try {
+        return objectMapper.readValue(bytes, Object.class);
+      } catch (Exception e) {
+        throw new SerializationException(e);
+      }
+    }
+  }
+
+  @Before
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    final int numPartitions = 3;
+    final int replicationFactor = 1;
+    kafka.utils.TestUtils.createTopic(zkClient, topicName, numPartitions, replicationFactor,
+        JavaConversions.asScalaIterable(this.servers).toSeq(),
+        new Properties());
+  }
+
+  @Override
+  protected String getEmbeddedContentType() {
+    return Versions.KAFKA_V1_JSON_JSON;
+  }
+
+  private final List<JsonTopicProduceRecord> topicRecordsWithKeys = Arrays.asList(
+      new JsonTopicProduceRecord("key", "value", 0),
```

Comment:
Same as in consumer test, we should probably test other types. We ran into this issue already (see #72) so it's probably best to make sure we have this covered in the JSON case with this initial commit.